### PR TITLE
docs: button component

### LINKED ISSUE #33: Ibimina - Button Component with Class, Text, and Click Function Props
###  Button Component with Class, Text, and Click Function Props
A reusable button component  that has the ability to accept class, text, and click function props. This would allow to easy customize the appearance, content, and behavior of the button component based on specific requirements.

### PR DESCRIPTION
**Pull request by ibimina**
## What type of PR is this? (check all applicable)

- [ ] Feature
- [ ] Refactor
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [x] Other (please specify here: Component creation)
  
  
## Description of what this Pull Request does
Added a CustomizableButton component in the general folder to the components folder

- The component should accept a className prop(customClass) which allows developers to pass custom CSS classes to style the button according to their needs.
- The component should accept a text prop(text) which allows developers to set the text content of the button.
- The button component should accept a onClick prop(onClickProp) which allows developers to pass a callback function that will be executed when the button is clicked.
- The button component should trigger the callback function passed via the onClick prop when the button is clicked.
- The button component should display the text content passed via the text prop.
- The button component should apply the CSS classes passed via the className prop to its DOM element.

## Related Tickets & Documents
- Closes  #33
- Related Issues: (enter ticket numbers here)
  
## QA Instructions, Screenshots, Recordings
(In this section, describe how to QA/review your PR. Feel free to use screenshots and/or short videos)
  
## Added/updated tests?
- [ ] Yes
- [ ] Not Yet, and here is why: __explain here__
- [ ] I need help with tests
- [x] No, and here is why: __explain here__

### CODE REVIEW

File: src/components/common/Button.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+function Button({text}) {
+    return ( <button>{text}</button>);
```
Comment (jjleigh) [inline src/components/common/Button.js:2]: Not a big deal but I would remove the white space after the (

File: src/components/common/Button.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+}
+
+export default Button;
```
Comment (jjleigh) [inline src/components/common/Button.js:5]: The symbol below this line indicates that there is no newline at the end of this file. It's not a big deal but to remove this you can just click enter after the ; and then commit it.

File: src/components/common/Button.js (hunk 1 above)
Comment (ibimina) [reply src/components/common/Button.js]: done

File: src/components/common/Button.js (hunk 2 above)
Comment (ibimina) [reply src/components/common/Button.js]: done

**Review by annonymousauthority — CHANGES_REQUESTED** [review]

File: src/components/common/Button.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+function Button({text}) {
+    return (<button>{text}</button>);
```
[the remark this answers is not in the document]
Comment (ibimina) [reply src/components/common/Button.js:2]: Noted

[the remark this answers is not in the document]
Comment (ibimina) [reply src/components/common/Button.js]: @annonymousauthority kindly review

**Review by msarit — APPROVED** [review]
I am going to defer to our React experts @jjleigh and @annonymousauthority but this looks good to me! 🎉

File: src/components/common/Button.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (jjleigh) [reply src/components/common/Button.js]: Good point about the component name!

**Review by jjleigh — APPROVED** [review]
The updates look good

**Review by annonymousauthority — APPROVED** [review]
LGTM

**Pull request merged**